src/core/compile.js

```diff
@@ -7,7 +7,7 @@ export const compile = (str, defs, data) => {
     return str.reduce((out, next, i) => {
         let tail = defs[i];
     
-        if (typeof defs[i] == "function") {
+        if (typeof defs[i] === "function") {
```

Comment:
Hey @cbbfcd, thanks for taking the time to open this! Really appreciate it!

The use of double equal, it's meant to keep the size low while maintaining proper functionality. Have you encountered issues with the double and not the triple, type checking, as well?

Reply:
@cristianbote thx for your reply!

I fully understand the intention of using double equality, which is also safe in this scenario,
So [here](https://github.com/cristianbote/goober/blob/master/src/core/parse.js#L23) is a similar.

and it's can be replaced with `typeof val == 'object'`.

Reply:
in fact, I prefer to use ===, the file size can be stolen from elsewhere

Reply:
Hey @cbbfcd so, I've managed to shave `14B` and that meant to get rid of this exact check. There's the other one you've mentioned. If you'd like you can change that one instead.

Reply:
i will make a new PR about it. thx~